kaggle-classification/tf_trainer/text_preprocessor.py

```diff
@@ -0,0 +1,85 @@
+"""Text Preprocessor."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl import app
+from absl import flags
+
+import nltk
+import numpy as np
+import tensorflow as tf
+from tf_trainer import types
+from typing import Tuple, Dict, Optional, List
+
+FLAGS = flags.FLAGS
+
+
+class TextPreprocessor():
+  """Text Preprocessor.
+
+  Takes an embedding and uses it to produce a word to index mapping and an
+  embedding matrix.
+  """
+
+  def __init__(self, embeddings_path: types.Path) -> None:
+    self._word_to_idx, self._embeddings_matrix, self._unknown_token = TextPreprocessor._get_word_idx_and_embeddings(
+        embeddings_path)  # type: Tuple[Dict[str, int], np.ndarray, int]
+
+  def word_to_idx_table(self) -> tf.contrib.lookup.HashTable:
+    """Get word to index mapping as a TF HashTable."""
+
+    keys = list(self._word_to_idx.keys())
+    values = list(self._word_to_idx.values())
+    table = tf.contrib.lookup.HashTable(
+        tf.contrib.lookup.KeyValueTensorInitializer(keys, values),
+        self._unknown_token)
+    return table
+
+  def word_embeddings(self, trainable=False) -> tf.Variable:
+    """Get word embedding TF Variable."""
+
+    embeddings_shape = self._embeddings_matrix.shape
+    initial_embeddings_matrix = tf.constant_initializer(self._embeddings_matrix)
+    embeddings = tf.get_variable(
+        name='word_embeddings',
+        shape=embeddings_shape,
+        initializer=initial_embeddings_matrix,
+        trainable=trainable)
+    return embeddings
+
+  @staticmethod
+  def _get_word_idx_and_embeddings(embeddings_path: types.Path,
```

Comment:
this code seems familiar - can we branch or make a library?

Reply:
My hope is to eventually deprecate the other versions